src/files_actions/viewerAction.ts

```diff
@@ -37,7 +37,7 @@ async function execAction(node: Node, view: View, dir: string): Promise<boolean|
 	}
 
 	pushToHistory(node, view, dir)
-	window.OCA.Viewer.open({ path: node.path, onPrev: pushToHistory, onNext: pushToHistory, onClose })
+	window.OCA.Viewer.open({ path: node.path, onPrev(fileInfo) { pushToHistory(fileInfo, view, dir) }, onNext(fileInfo) { pushToHistory(fileInfo, view, dir) }, onClose })
```

Comment:
So make it better readable?

```suggestion
	window.OCA.Viewer.open({
		path: node.path,
		onPrev(fileInfo) {
			pushToHistory(fileInfo, view, dir)
		},
		onNext(fileInfo) {
			pushToHistory(fileInfo, view, dir)
		},
		onClose,
	})
```

Reply:
You're right, it is time :)